tests/unit/specs/components/skeletons/NavBar.spec.js

```diff
@@ -0,0 +1,245 @@
+    expect(true).toBe(true)
+  })
+
+  test('Rendered correctly', () => {
+    initComponent()
+    expect(wrapper.isVueInstance()).toBe(true)
+  })
+
+  test('computed name', () => {
+    initComponent()
+    expect(wrapper.vm.name).toEqual('')
+  })
+
+  test('computed name', () => {
+    initComponent()
+    store.state.currentUser = {
+      name: 'Karnando Sepryan'
+    }
+    expect(wrapper.vm.name).toEqual('Karnando Sepryan')
+  })
+
+  test('computed firstName > 1 word', () => {
+    initComponent()
+    store.state.currentUser = {
+      name: 'Karnando Sepryan'
+    }
+    expect(wrapper.vm.firstName).toEqual('Karnando')
+  })
+
+  test('computed firstName one word', () => {
+    initComponent()
+    store.state.currentUser = {
+      name: 'Karnando'
+    }
+    expect(wrapper.vm.firstName).toEqual('Karnando')
+  })
+
+  test('login not logged in', () => {
+    initComponent()
+    wrapper.vm.$router.push = jest.fn()
+    wrapper.vm.login()
+    expect(wrapper.vm.$router.push).toHaveBeenCalledWith({ name: 'login' })
+  })
+
+  test('login already logged in', () => {
+    initComponent()
+    store.state.currentUser = {
+      name: 'Karnando Sepryan'
+    }
+    wrapper.vm.login()
+  })
+
+  test('logout', () => {
+    initComponent()
+    const spy = jest.spyOn(wrapper.vm, 'attemptLogout')
+    wrapper.vm.logout()
+    expect(spy).toHaveBeenCalledTimes(1)
+  })
+
+  test('successAttemptLogout', () => {
+    initComponent()
+    wrapper.vm.$router.push = jest.fn()
+    wrapper.vm.successAttemptLogout()
+    expect(wrapper.vm.$cookies.remove).toHaveBeenCalledTimes(1)
+    expect(wrapper.vm.$router.push).toHaveBeenCalledWith({ name: 'feeds' })
+  })
+
+  test('goToProfile', () => {
+    initComponent()
+    wrapper.vm.$router.push = jest.fn()
+    wrapper.vm.goToProfile()
+    expect(wrapper.vm.$router.push).toHaveBeenCalledWith({ name: 'profile' })
+  })
+
+  test('goToNotifications case 1', () => {
+    initComponent()
+    wrapper.vm.$router.push = jest.fn()
+    wrapper.vm.goToNotifications()
+    expect(wrapper.vm.$router.push).toHaveBeenCalledWith({ name: 'notifications' })
+  })
+
+  // test('goToNotifications case 2', () => {
+  //   window.location.reload = jest.fn()
+  //   const $route = {
+  //     name: 'notifications'
+  //   }
+  //   const $router = {
+  //     push: jest.fn()
+  //   }
+  //   const lv = createLocalVue()
+  //   lv.use(Vuex)
+  //   store = initStore()
+  //   wrapper = shallowMount(NavBar, {
+  //     mocks: {
+  //       $route,
+  //       $router
+  //     },
+  //     store,
+  //     localVue: lv,
+  //     stubs: [
+  //       'BaseCard',
+  //       'BaseButton',
+  //       'BaseInput',
+  //       'BaseSelect',
+  //       'v-date-picker',
+  //       'font-awesome-icon'
+  //     ],
+  //     sync: false
+  //   })
+  //   wrapper.vm.goToNotifications()
+  //   expect(wrapper.vm.$router.push).toHaveBeenCalledTimes(0)
+  //   expect(window.location.reload).toHaveBeenCalled()
+  // })
```

Comment:
Can't this be removed?

Reply:
itu sbenernya kodingannya agung ricky, cuma gatau kenapa fail, belom sempet gua fix wkwk

Reply:
done ya